internal/comm/httptransport_test.go

```diff
@@ -391,6 +399,339 @@ func TestNewHTTPTransport_TLS_FilePathFailure(t *testing.T) {
+	_, err = tr1.PullBlocks(timeout1, 1, 5, 0)
+	require.EqualError(t, err, "PullBlocks canceled: context deadline exceeded")
+	timeout2, _ := context.WithTimeout(context.Background(), 500*time.Millisecond)
+	_, err = tr2.PullBlocks(timeout2, 1, 5, 0)
+	require.EqualError(t, err, "PullBlocks canceled: context deadline exceeded")
+
+	// emulate a config tx update to change the endpoint of server 3
+	err = tr1.UpdatePeers(sharedConfigUpdated.ConsensusConfig.Members[2:], nil, nil, sharedConfigUpdated)
+	require.NoError(t, err)
+	err = tr2.UpdatePeers(sharedConfigUpdated.ConsensusConfig.Members[2:], nil, nil, sharedConfigUpdated)
+	require.NoError(t, err)
+
+	// messages are received
+	tr1.SendConsensus([]raftpb.Message{{To: 3}})
+	tr1.SendConsensus([]raftpb.Message{{To: 3}})
+	require.Eventually(t,
+		func() bool {
+			return cl3.ProcessCallCount() == 2
+		},
+		10*time.Second, 10*time.Millisecond,
+	)
+	tr2.SendConsensus([]raftpb.Message{{To: 3}})
+	tr2.SendConsensus([]raftpb.Message{{To: 3}})
+	tr2.SendConsensus([]raftpb.Message{{To: 3}})
+	require.Eventually(t,
+		func() bool {
+			return cl3.ProcessCallCount() == 5
+		},
+		10*time.Second, 10*time.Millisecond,
+	)
+
+	// PullBlocks will succeed
+	timeout1, _ = context.WithTimeout(context.Background(), 10*time.Second)
+	blocks, err := tr1.PullBlocks(timeout1, 1, 5, 0)
+	require.NoError(t, err)
+	require.Len(t, blocks, 5)
+	timeout2, _ = context.WithTimeout(context.Background(), 10*time.Second)
+	blocks, err = tr2.PullBlocks(timeout2, 1, 5, 0)
+	require.NoError(t, err)
+	require.Len(t, blocks, 5)
+}
+
+// Scenario: remove peers.
+// - generate a test configuration for 3 servers and start them.
+// - check connectivity to the 3rd server.
+// - emulate a config tx to remove the 3rd server; check connectivity to it.
+func TestHTTPTransport_RemovePeers(t *testing.T) {
+	lg, err := logger.New(
+		&logger.Config{
+			Level:         "info",
+			OutputPath:    []string{"stdout"},
+			ErrOutputPath: []string{"stderr"},
+			Encoding:      "console",
+		},
+	)
+	require.NoError(t, err)
+
+	localConfigs, sharedConfig := newTestSetup(t, 3)
+
+	cl1 := &mocks.ConsensusListener{}
+	tr1, _ := comm.NewHTTPTransport(&comm.Config{
+		LocalConf:    localConfigs[0],
+		Logger:       lg,
+		LedgerReader: &mocks.LedgerReader{},
+	})
+	require.NotNil(t, tr1)
+	err = tr1.SetConsensusListener(cl1)
+	require.NoError(t, err)
+	err = tr1.SetClusterConfig(sharedConfig)
+	require.NoError(t, err)
+
+	cl2 := &mocks.ConsensusListener{}
+	tr2, _ := comm.NewHTTPTransport(&comm.Config{
+		LocalConf:    localConfigs[1],
+		Logger:       lg,
+		LedgerReader: &mocks.LedgerReader{},
+	})
+	require.NotNil(t, tr1)
+	err = tr2.SetConsensusListener(cl2)
+	require.NoError(t, err)
+	err = tr2.SetClusterConfig(sharedConfig)
+	require.NoError(t, err)
+
+	err = tr1.Start()
+	require.NoError(t, err)
+	defer tr1.Close()
+
+	err = tr2.Start()
+	require.NoError(t, err)
+	defer tr2.Close()
```

Comment:
This code snippet is repeated thrice (50 lines of code). I know that for this test, we start 3 instead of 2 (but other tests use 2). We can reduce some code duplication if we use table based test method. If that is challenging, we can at least use a function to create a setup struct and reuse it.

Reply:
Done. I am reusing the `startTransportWithLedger` helper method which removes all that duplication.